zilliqa/src/contracts/deposit.sol

```diff
@@ -138,61 +207,238 @@ contract Deposit {
         require(peerId.length == 38);
         require(signature.length == 96);
 
-        require(_stakerKeys.length < _maximumStakers, "too many stakers");
+        require(committeeKeys.length < maximumStakers, "too many stakers");
 
         // Verify signature as a proof-of-possession of the private key.
         bool pop = _popVerify(blsPubKey, signature);
         require(pop, "rogue key check");
 
-        uint256 keyIndex = _stakersMap[blsPubKey].keyIndex;
-        if (keyIndex == 0) {
-            // The staker will be at index `_stakerKeys.length`. We also need to add 1 to avoid the 0 sentinel value.
-            _stakersMap[blsPubKey].keyIndex = _stakerKeys.length + 1;
-            _stakerKeys.push(blsPubKey);
+        // If staker info isn't stored yet, do so
+        // peerId is guaranteed to be non-zero for all stakers
```

Comment:
Is it or do have have to check, and even if it is, why is the comment relevant here?